cerberus/util.py

```diff
@@ -9,3 +12,40 @@ def throw_if_bad_response(response):
     except RequestException as e:
         msg = 'Response code: {}; response body:\n{}'.format(response.status_code, json.dumps(response.json(), indent=2))
         raise CerberusClientException(msg)
+
+
+def get_with_retry(url, retry=3, **kwargs):
+    return request_with_retry(url, 'get', retry, **kwargs)
+
+
+def post_with_retry(url, retry=3, **kwargs):
+    return request_with_retry(url, 'post', retry, **kwargs)
+
+
+def put_with_retry(url, retry=3, **kwargs):
+    return request_with_retry(url, 'put', retry, **kwargs)
+
+
+def delete_with_retry(url, retry=3, **kwargs):
+    return request_with_retry(url, 'delete', retry, **kwargs)
+
+
+def head_with_retry(url, retry=3, **kwargs):
+    return request_with_retry(url, 'head', retry, **kwargs)
+
+
+def request_with_retry(url, verb, retry, **kwargs):
+    request = {'get': requests.get,
+               'post': requests.post,
+               'put': requests.put,
+               'delete': requests.delete,
+               'head': requests.head}
+    resp = None
+    for retry_attemp_number in range(retry):
+        resp = request[verb](url, **kwargs)
+        if not resp.status_code >= 500:
+            return resp
+        else:
+            # exponential backoff
+            time.sleep(2**retry_attemp_number)
```

Comment:
This seems like a lot of time. I would start from some number of milliseconds.

Amazon for example starts from 100 millis:
https://docs.aws.amazon.com/general/latest/gr/api-retries.html